sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1543,6 +1544,90 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Gets all eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible application that can be requested for
+   * application offering change.
+   */
+  async getEligibleApplicationOfferingChangeRecords(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResults<Application>> {
+    const applicationQuery = this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "application.id",
+        "currentAssessment.id",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "student.id",
+        "user.firstName",
+        "user.lastName",
+      ])
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "application.applicationOfferingChangeRequest",
+        "applicationOfferingChangeRequest",
+      )
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .where(
+        new Brackets((qb) =>
+          qb
+            .where("applicationOfferingChangeRequest.id IS NULL")
+            .orWhere(
```

Comment:
Can you please explain why we need this applicationOfferingChangeRequestStatus Or condition? Is that to show the applications that are already approved or declined? If that is the case, IMO we can just change the condition of Id and add an or to have the where with status approved or denied

Reply:
This query is for the `available for request` applications list,  ![image](https://github.com/bcgov/SIMS/assets/77353155/8862e917-643d-42ff-a659-7c031b9b8591)